src/tests/test_davidson.cpp

```diff
@@ -0,0 +1,98 @@
+
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_MODULE davidson_test
+
+
```

Comment:
I want an error handling test case as well. So what happens if it does not converge,how  many iterationsetc...

Reply:
added a test case that should fail (only one iteration)